### PR TITLE
OP-134/Sylius upgrade

### PR DESCRIPTION
**Pull request by rafalmal**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -131,7 +140,7 @@ jobs:
                 name: Prepare test application assets
                 run: |
                     (cd tests/Application && bin/console assets:install public -vvv)
-                    (cd tests/Application && yarn build)
+                    (cd tests/Application && yarn encore dev)
```
Comment (senghe) [inline .github/workflows/build.yml:143]: This change needs to be put in the installation docs

File: src/Bus/Dispatcher.php [hunk 2]
```diff
@@ -17,8 +17,8 @@ final class Dispatcher implements DispatcherInterface
 {
     use HandleTrait;
 
-    /** @var MessageBusInterface */
-    private $messageBus;
+//    /** @var MessageBusInterface */
```
Comment (senghe) [inline src/Bus/Dispatcher.php]: What's this?

File: src/Callback/PreserveOrderTokenUponRedirectionCallback.php [hunk 3]
```diff
@@ -11,22 +11,29 @@
...
+        if (null == $session->getMainRequest()) {
+            return;
+        }
+        $this->session = $session->getSession();
     }
 
     public function __invoke(OrderInterface $order): void
     {
+        if (null === $this->session) {
+            return;
```
Comment (senghe) [inline src/Callback/PreserveOrderTokenUponRedirectionCallback.php:35]: I would throw exception for better debugging

File: src/Processor/PaymentResponseProcessor/FallbackResponseProcessor.php [hunk 4]
```diff
@@ -14,6 +14,9 @@
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
 
+/**
+ * @psalm-suppress PropertyNotSetInConstructor
```
Comment (senghe) [inline src/Processor/PaymentResponseProcessor/FallbackResponseProcessor.php]: Why it's suppressed? If there is a reason, you've to describe it in a comment

File: src/Resolver/Order/PaymentCheckoutOrderResolver.php [hunk 5]
```diff
@@ -66,7 +65,7 @@ private function getCurrentOrder(): ?OrderInterface
             return null;
         }
         /**
-         * @psalm-suppress MixedReturnStatement
+         * @phpstan-ignore-next-line
```
Comment (senghe) [inline src/Resolver/Order/PaymentCheckoutOrderResolver.php]: We need the description, if it's super necessary.

File: tests/Application/config/sylius/1.12/bundles.php [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<?php
```
Comment (senghe) [inline tests/Application/config/sylius/1.12/bundles.php:1]: No OS header

File: tests/Application/config/sylius/1.12/packages/jms_serializer.yaml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+#jms_serializer:
```
Comment (senghe) [inline tests/Application/config/sylius/1.12/packages/jms_serializer.yaml]: What's this?

File: tests/Behat/Context/Ui/Admin/ManagingPaymentMethodAdyenContext.php [hunk 8]
```diff
@@ -68,7 +68,11 @@ public function adyenServiceWillConfirmMerchantAccountAndApiKeyAreValid(string $
         $this->kernel
             ->getContainer()
             ->get('tests.bitbag.sylius_adyen_plugin.behat.context.api_mock_client')
-            ->setJsonHandler(function (Service $service, string $url, array $payload) use ($merchantAccount, $apiKey) {
+            ->setJsonHandler(function (
+                Service $service,
```
Comment (senghe) [inline tests/Behat/Context/Ui/Admin/ManagingPaymentMethodAdyenContext.php:72]: Bad formatting

File: tests/HttpClientStub.php [hunk 9]
```diff
@@ -32,9 +32,12 @@ public function setPostHandler(?callable $postHandler): void
...
     }
 
-    public function requestJson(Service $service, $requestUrl, $params)
-    {
-        if (self::$jsonHandler !== null) {
+    public function requestJson(
+        Service $service,
+        $requestUrl,
+        $params
+    ) {
```
Comment (senghe) [inline tests/HttpClientStub.php]: Missing return and parameters types

File: tests/Unit/Resolver/Address/StreetAddressResolverTest.php [hunk 10]
```diff
@@ -1,4 +1,6 @@
-<?php declare(strict_types=1);
+<?php
```
Comment (senghe) [inline tests/Unit/Resolver/Address/StreetAddressResolverTest.php:1]: No OS header

**Review by senghe — CHANGES_REQUESTED** [review]

File: src/Bus/Dispatcher.php (hunk 2 above)
Comment (rafalmal) [reply src/Bus/Dispatcher.php]: Fixed in next commit

File: tests/HttpClientStub.php (hunk 9 above)
Comment (rafalmal) [reply tests/HttpClientStub.php]: This class implements Adyen/ClientInterface and there parameters are not typehinted so if i try to typehint them get an error.

File: src/Callback/PreserveOrderTokenUponRedirectionCallback.php (hunk 3 above)
Comment (rafalmal) [reply src/Callback/PreserveOrderTokenUponRedirectionCallback.php:35]: This callback is used while building fixtures, so we cant throw there.

File: src/Processor/PaymentResponseProcessor/FallbackResponseProcessor.php (hunk 4 above)
Comment (rafalmal) [reply src/Processor/PaymentResponseProcessor/FallbackResponseProcessor.php]: Removed psalm suppress, and added unused property to constructor in next commit.

**Review by senghe — APPROVED** [review]

**Pull request merged**